### PR TITLE
Fix newsfeed API

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR makes it possible to query the news feed from the API.

### CODE REVIEW

File: skyportal/handlers/api/news_feed.py [hunk 1]
```diff
@@ -44,13 +54,27 @@ def get(self):
...
-        preferences = (
-            self.current_user.preferences if self.current_user.preferences else {}
-        )
-        if 'newsFeed' in preferences and 'numItems' in preferences['newsFeed']:
-            n_items = min(int(preferences['newsFeed']['numItems']), 50)
+
+        if hasattr(self.current_user, 'preferences'):
+            preferences = (
+                self.current_user.preferences if self.current_user.preferences else {}
+            )
```
Comment (guynir42) [inline skyportal/handlers/api/news_feed.py:61]:
```suggestion
        if hasattr(self.current_user, 'preferences') and isinstance(self.current_user.preferences, dict):
            preferences = self.current_user.preferences
```

File: skyportal/handlers/api/news_feed.py [hunk 2]
```diff
@@ -44,13 +54,27 @@ def get(self):
...
         else:
-            n_items = 10
+            preferences = {}
+
+        n_items = self.get_query_argument("numItems", None)
+        if n_items is None:
+            if 'newsFeed' in preferences and 'numItems' in preferences['newsFeed']:
+                n_items = min(int(preferences['newsFeed']['numItems']), 50)
```
Comment (guynir42) [inline skyportal/handlers/api/news_feed.py:68]: I don't get where `50` comes from.

File: skyportal/tests/api/test_newsfeed.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            "ra": 235.22,
+            "dec": -23.33,
+            "redshift": 3,
+            "group_ids": [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+
+    params = {'numItems': 100}
```
Comment (guynir42) [inline skyportal/tests/api/test_newsfeed.py:24]: What happens if there are more than 100 sources in the test DB because of previous tests?

**Review by guynir42 — COMMENTED** [review]
Minor comments/suggestions.

File: skyportal/handlers/api/news_feed.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/news_feed.py:68]: This was the original default

File: skyportal/tests/api/test_newsfeed.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/tests/api/test_newsfeed.py:24]: Then this would fail. I can bump this to 1000.

File: skyportal/handlers/api/news_feed.py (hunk 2 above)
Comment (guynir42) [reply skyportal/handlers/api/news_feed.py:66]: Shouldn't this number be either the new default or the new maximum or something that's somehow documented?

Comment (mcoughlin) [reply skyportal/handlers/api/news_feed.py]: yeah i am setting this to a default at the top of the file.

**Review by guynir42 — APPROVED** [review]

File: skyportal/handlers/api/news_feed.py [hunk 4]
```diff
@@ -44,13 +55,29 @@ def get(self):
               application/json:
                 schema: Error
         """
-        preferences = (
-            self.current_user.preferences if self.current_user.preferences else {}
-        )
-        if 'newsFeed' in preferences and 'numItems' in preferences['newsFeed']:
-            n_items = min(int(preferences['newsFeed']['numItems']), 50)
+
+        if hasattr(self.current_user, 'preferences') and isinstance(
```
Comment (stefanv) [inline skyportal/handlers/api/news_feed.py:59]:
```
preferences = getattr(self.current_user, 'preferences', None) or {}
```

File: skyportal/handlers/api/news_feed.py [hunk 5]
```diff
@@ -44,13 +55,29 @@ def get(self):
...
+            self.current_user.preferences, dict
+        ):
+            preferences = self.current_user.preferences
         else:
-            n_items = 10
+            preferences = {}
+
+        n_items = self.get_query_argument("numItems", None)
```
Comment (stefanv) [inline skyportal/handlers/api/news_feed.py:66]: This feels too complicated.
How about getting:
1. arg n_items
2. preference n_items
3. choose and compare

File: skyportal/tests/api/test_newsfeed.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            "dec": -23.33,
+            "redshift": 3,
+            "group_ids": [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+
+    params = {'numItems': 1000}
+    status, data = api('GET', 'newsfeed', token=view_only_token, params=params)
```
Comment (stefanv) [inline skyportal/tests/api/test_newsfeed.py:25]: No check for erroneous number of items requested

**Pull request merged**